frontend/javascripts/admin/dataset/dataset_upload_view.js

```diff
@@ -89,6 +91,27 @@ class DatasetUploadView extends React.PureComponent<PropsWithForm, State> {
           isUploading: true,
         });
 
+        const beforeUnload = action => {
+          // Only show the prompt if this is a proper beforeUnload event from the browser
+          // or the pathname changed
+          // This check has to be done because history.block triggers this function even if only the url hash changed
+          if (action === undefined || evt.pathname !== window.location.pathname) {
+            const { isUploading } = this.state;
+            if (isUploading) {
+              window.onbeforeunload = null; // clear the event handler otherwise it would be called twice. Once from history.block once from the beforeunload event
+              window.setTimeout(() => {
+                // restore the event handler in case a user chose to stay on the page
+                window.onbeforeunload = beforeUnload;
+              }, 500);
+              return messages["save.leave_page_unfinished"];
```

Comment:
Could you create a new message string for this scenario? Right now, it says: `WARNING: You have unsaved progress that may be lost when hitting OK. Please click cancel, wait until the progress is saved and the save button displays a checkmark before leaving the page..`. However, the text doesn't fit the dataset upload page :)

Reply:
I totally forgot about that, thanks for pointing it out! It's adjusted now.